### ISSUE TITLE
[Feature Request] Add ability to modify .model-dialog and .modal-content 

### ISSUE THREAD

**Issue by sumeetattree** [kaliber5/ember-bootstrap#469]
Currently there is no way to set classes on `.modal-dialog` and `.modal-content`. It would be nice to be able to modify these elements.

Use cases:
* Make a full-width, full-height modal
  - classes that need modification `.modal-dialog` and `.modal-content`
* Custom modal positioning
  - classes that need modification `.modal-dialog`
* Customize the modal background color (`.modal-content`)
  - classes that need modification `.modal-content`

Presently one has to override the `.modal .modal-dialog` and `.modal .modal-content` in stylesheets to get this to work.

I propose:
Approach 1:
```handlebars
{{#bs-modal dialogOptions=(hash class="bg-dark" other="property") as |modal|}}
{{/bs-modal}}
```

Approach 2: Or alternatively expose `bs-modal-dialog` and `bs-modal-content` as helpers. These components will just serve as wrappers for us to apply classes onto them. This enables us to do:
```handlebars
{{#bs-modal as |modal|}}
  {{#modal.dialog}}
    {{#modal.content}}
      {{#modal.header}}{{/modal.header}}
      {{#modal.body}}{{/modal.body}}
      {{#modal.footer}}{{/modal.footer}}
    {{/modal.content}}
  {{/modal.dialog}}
{{/bs-modal}}
```
Personally I favor the Approach 2. But I can see merits in using Approach 1 as it reduces boilerplate code.
I would like to hear your feedback on these. 

And lastly, thank you for this awesome library!

**Comment by simonihmig** [comment]
There is already a possibility to set additional classes to `.modal-dialog`. The `class` property of `bs-modal` is piped through to the `bs-modal/dialog` component: https://github.com/kaliber5/ember-bootstrap/blob/master/addon/templates/components/common/bs-modal.hbs#L11

This needs to be better documented though!

Given that you can set the class on the `.modal-dialog` I don't think you need to also set one to the `.modal-content`, as you can target it like `.modal-dialog.my-custom-class .modal-content`.

So, would that be enough to cover your needs?

**Comment by srvance** [comment]
@simonihmig beat me to it.

In addition to his suggestions, the Bootstrap classes are designed to be modified, so if it's a change you want across your entire application, you can just modify the classes.

Another common pattern is to specialize the Bootstrap classes under a well known ID or other scope, often by putting an application-specific ID on one of the outermost tags in your markup so you can specialize the usage like `#myapplication .modal-dialog { ... }`.

Do these approaches work for you?

**Comment by sumeetattree** [comment]
@simonihmig You are correct in that it needs to be documented. [Looking at this line](https://github.com/kaliber5/ember-bootstrap/blob/9141c0e1ace824c21fceb10f782806f43effc5cf/addon/templates/components/common/bs-modal/dialog.hbs#L1), it's only passing the `{{sizeClass}}`, it didn't occur to me immediately that I can pass classes to the `size` property. But in my opinion that's not very intuitive and doesn't feel right.

@srvance 
> In addition to his suggestions, the Bootstrap classes are designed to be modified, so if it's a change you want across your entire application, you can just modify the classes.

This is how I am doing it currently. These are overriden at the stylesheet level in my project. However, this didn't feel very bootstrap-y to me, in the sense that I couldn't just compose the modals from existing classes. 

The reason why I suggest these changes is to bring parity between `ember-bootstrap` and `bootstrap` api. Having a shortcut to remove boilerplate code is great but losing the flexibility is downside as well.

That said it might just be a very narrow use case not worth exploring, in which case please feel free to close this issue.
